### PR TITLE
Integrating Cloud Firestore database in Android-Kotlin

### LINKED ISSUE #2407: How to integrate cloud Firestore database in Android using Kotlin
### Proposed title of article

How to integrate cloud Firestore database in Android using Kotlin

### Introduction paragraph (2-3 paragraphs):

Firestore is a cloud-hosted NoSQL database that can be integrated into various platforms(iOS, Android, and web apps) via the respective native  SDKs. It's a reliable database in that it offers flexible data storage structures expressive querying, offline support, improved identity and access management(IAM), and synchronized real-time updates.
In this article, we'll learn how Firestore can be used in an Android app using the Kotlin programming language. Modern apps are expected to run smoothly and offer the best possible user experience and that's exactly what Firestore brings us.

### Key takeaways: 

This article will guide the reader to:
1. Understand why we need a cloud-hosted database.
2. Implement the Firestore database in an Android app.
3. Perform CRUD operations using the developer console.
4. Manage data access rules/permissions.

### References: 
N/A

### PR DESCRIPTION
**Pull request by nonimdiana**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2407

### CODE REVIEW

File: content/articles/integrating-firestore-in-android-kotlin/index.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+### Tutorial Prerequisites
+To follow through this article, you'll need to have a basic experience in:
+
+- Using Kotlin.
+- General usage of [Android Studio](https://developer.android.com/studio)
+- View binding or data binding
+- Managing a Google account.
+
+### Table of contents
```
Comment (kayere) [inline content/articles/integrating-firestore-in-android-kotlin/index.md:29]: Please add the links to the subtitles.

File: content/articles/integrating-firestore-in-android-kotlin/index.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+- Upload and Read data.
+- Conclusion.
+
+### How Firestore stores data
+Unlike other popular databases, Firestore stores data in a [NoSQL format](https://youtu.be/v_hR4K4auoQ), that is, data is stored in JSON objects within nodes. These nodes are referred to as documents. A document can not exist on its own. There must be a parent container that holds at least one document. This container is referred to as a collection. As the database grows, a tree-like structure is formed with the related documents and collections connected to a common collection with the aid of their respective `IDs`. Firestore is unique in that it supports creation of collections inside documents. These buried collections are known as sub-collections.
+
+For instance, to create a `users` database, we'll create a collection called `users` and add documents with `unique` IDs each holding the user's attributes. 
+These attributes can be of the following data types.
+
+### Datatypes that Firestore supports
```
Comment (kayere) [inline content/articles/integrating-firestore-in-android-kotlin/index.md:43]: You could just add this to the previous subtitle.

**Review by kayere — COMMENTED** [review]
Hello @nonimdiana - welcome to section's engineering education program. This is a great first article, very well written and easy to follow through. Please check the suggestions I have made before we proceed to final review.

**Comment by nonimdiana** [comment]
Hey @kayere, Thank you for welcoming me to this program. I've made the changes you requested. Kindly proceed with the review.

**Review by kayere — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by ahmadmardeni1** [comment]
Welcome to Section, and congrats on your first tutorial! 🎉 @nonimdiana 

Good work @nonimdiana @kayere

**Pull request merged**